lib/versioning/cargo/index.ts

```diff
@@ -96,7 +104,7 @@ function getNewValue({
     currentVersion,
     newVersion,
   });
-  let newCargo = npm2cargo(newSemver);
+  let newCargo = npm2cargo(newSemver ?? '');
```

Comment:
This looks wrong to me.

Reply:
Error: newSemver has type string | null while `npm2cargo` allows only string.

Solution: Fallback to an empty string when newSemver is null. An empty string should work just fine cause:
https://github.com/renovatebot/renovate/blob/2b78fa575e145edecf0c491320f2776ba72e9bab/lib/versioning/cargo/index.ts#L37-L40

In the case of falsy input, it returns the same input which is then used as `newCargo`.
```js
  let newCargo = npm2cargo(newSemver);
  // istanbul ignore if
  if (!newCargo) {
    logger.info(
      { currentValue, newSemver },
      'Could not get cargo version from semver'
    );
    return currentValue;
  }
  ```

Reply:
```suggestion
  let newCargo = newSemver ? npm2cargo(newSemver): null;
```

This looks cleaner to me. Maybe `npm2cargo` should be widen to allow null values